interfaces/connection.go

```diff
@@ -57,6 +57,18 @@ type Attrer interface {
 	Lookup(path string) (value interface{}, ok bool)
 }
 
+type SnippetKey struct {
```

Comment:
Perhaps something like this:
```suggestion
// SnippetKey is an opaque string identifying a class of snippets.
//
// Some APIs require the use of snippet keys to allow adding many different snippets
// with the same key but possibly different priority.
type SnippetKey struct {
```